docs/src/userguide/memoryviews.rst

```diff
@@ -691,6 +691,49 @@ reject None input straight away in the signature, which is supported in Cython
 Unlike object attributes of extension classes, memoryview slices are not
 initialized to None.
 
+
+Pass data from a C function via pointer
+=======================================
+
+Since use of pointers in C is ubiquitous, here we give a quick example of how
+to call C functions whose arguments contain pointers. Let's suppose you want to
+manage an array (allocate and deallocate) with NumPy (it can also be Python arrays, or
+anything that supports the buffer interface), but you want to perform computation on this
+array with an external C function implemented in :file:`C_func_file.c`:
+
+.. literalinclude:: ../../examples/memoryviews/C_func_file.c
+    :linenos:
+
+This file have a header file called :file:`C_func_file.h` containing::
+
+    void multiply_by_10_in_C(double arr[], unsigned int n);
+
+where ``arr`` points to the array and ``n`` is its size.
+
+You can call the function in a Cython file in the following way:
+
+.. literalinclude:: ../../examples/memoryviews/memview_to_c.pyx
+    :linenos:
+
+Several things to note:
+
+ - ``&arr_memview[0]`` can be understood as 'the adress of the first element of the
+   memoryview'.
```

Comment:
I think it would be helpful to add: "For contiguous arrays, this is equivalent to the start address of the flat memory buffer."